modules/govuk_elasticsearch/manifests/backup.pp

```diff
@@ -39,71 +34,53 @@
 # [*cron_minute*]
 #   The minute that the job should run
 #
-# === Variables
-#
-# [*json_es_indices*]
-#   Formatted indexes to be backed up when script is invoked
-#
-class govuk_elasticsearch::backup(
-  $aws_access_key_id,
-  $aws_secret_access_key,
+class govuk_elasticsearch::backup (
+  $ensure = 'present',
+  $aws_access_key_id = undef,
+  $aws_secret_access_key = undef,
   $aws_region = 'eu-west-1',
-  $s3_bucket,
-  $aws_source_file = '/etc/awspasswd',
-  $user = 'govuk-backup',
+  $s3_bucket = undef,
   $es_repo = undef,
   $es_indices = [],
   $cron_hour = 0,
   $cron_minute = 0,
   $alert_hostname = 'alert.cluster',
 ){
 
-    $threshold_secs = 28 * 3600
-    $service_desc = 'Elasticsearch-Index_Backup'
-    $json_es_indices = join(regsubst($es_indices, '(.*)', '"\1"'), ',')
-
-    include ::backup::client
-    include govuk_elasticsearch::housekeeping
-
-    file { $aws_source_file:
-      ensure  => present,
-      owner   => $user,
-      group   => $user,
-      mode    => '0640',
-      content => template('govuk_elasticsearch/awspasswd.erb'),
-    }
+  $threshold_secs = 28 * 3600
+  $service_desc = 'Elasticsearch-Index_Backup'
+  $json_es_indices = join(regsubst($es_indices, '(.*)', '"\1"'), ',')
 
+  if $s3_bucket {
```

Comment:
If all the resources of the class require `s3_bucket` being defined, why don't we make the parameter mandatory in the class definition?

Reply:
Good point.